app/battle/battle.py

```diff
@@ -0,0 +1,55 @@
+def apply_armor(knight: dict) -> int:
+    return sum(piece["protection"] for piece in knight["armour"])
+
+
+def apply_weapon(knight: dict) -> int:
+    return knight["power"] + knight["weapon"]["power"]
+
+
+def apply_potion(knight: dict) -> None:
+    if knight["potion"] is not None:
+        effects = knight["potion"]["effect"]
+        if "power" in effects:
+            knight["power"] += effects["power"]
+        if "protection" in effects:
+            knight["protection"] += effects["protection"]

```

Comment:
The 'protection' attribute is being updated here, but it is initially set by the 'apply_armor' function. Ensure that 'protection' is intended to be modified by potions, as this might overwrite the armor's protection value.